backend/routes/project.ts

```diff
@@ -265,26 +278,42 @@ export async function modProject(
         description: checkedId.description,
     });
 
-    if (checkedId.modifyRoles !== undefined) {
-        for (const changeRolePositions of checkedId.modifyRoles.roles) {
-            const foundRole = await ormRole.getRole(changeRolePositions.id);
-            if (foundRole !== null) {
-                await ormPrRole.updateProjectRole({
-                    projectRoleId: changeRolePositions.id,
+    const oldProjectRoles = await ormPrRole.getProjectRoleNamesByProject(
+        checkedId.id
+    );
+
+    if (checkedId.roles !== undefined) {
+        loop1: for (const role of checkedId.roles.roles) {
+            for (const projectRole of oldProjectRoles) {
+                if (role.name === projectRole.role.name) {
+                    if (role.positions !== projectRole.positions) {
+                        if (role.positions === 0) {
+                            await ormPrRole.deleteProjectRole(
+                                projectRole.role_id
+                            );
+                        } else {
+                            await ormPrRole.updateProjectRole({
+                                projectRoleId: projectRole.project_role_id,
+                                projectId: checkedId.id,
+                                roleId: projectRole.role_id,
+                                positions: role.positions,
+                            });
+                        }
+                        continue loop1;
```

Comment:
Use `break;` instead of jumps

Reply:
I don't think a break will work because i want to continue executing the fist loop, not execute the code after the second loop and then continue executing the first loop.

Reply:
`break` only exits the innermost loop: (using `node` REPL)
```js
for(let i of [1,2,3,4,5,6]) {
  for(let j of [11,12,13,14,15]) {
    if(j > 12) break;
    console.log('j is ' + j);
  }
  console.log('i is ' + i);
}
```
-> Result:
```
j is 11
j is 12
i is 1
j is 11
j is 12
i is 2
j is 11
j is 12
i is 3
j is 11
j is 12
i is 4
j is 11
j is 12
i is 5
j is 11
j is 12
i is 6
```
(as you can see, the outermost loop continues (1 ... 6), while the innermost is halted each time after the 12)

Reply:
Yes but in this example, i wouldn't want to execute the console.log in the outer loop anymore when a certain condition in the inner loop is met.

Reply:
In that case (my bad), I still think that a normal `break` with a sentinel (bool) would be cleaner.

Reply:
Yeah that's true, i'll rewrite the code!